src/java.base/share/classes/sun/security/ssl/ServerHello.java

```diff
@@ -745,10 +761,48 @@ private static CipherSuite chooseCipherSuite(
             }
 
             // no cipher suites in common
+            printServerEnabledCipherSuites(shc, null, null);
```

Comment:
I'd suggest removing the debug flag checks from line 776 and wrapping each call of `printServerEnabledCipherSuites` with the debug flag check instead - it's more obvious to reader IMO.

Reply:
Updated the code, see new commit.